### PR TITLE
Adding logging to DDNS Module

### PR DESCRIPTION
**Pull request by DeForce**
### What does this PR do?
This PR adds logging to DDNS Module so debugging is easier

### What issues does this PR fix or reference?
#52648 

### Previous Behavior
No logging whatsoever, only thing that is happening is a return boolean (True, False, None)

### New Behavior
Now we have varying degrees of logging. Warning, Info and Debug logging is implemented.

### Tests written?
No

### Commits signed with GPG?
No

### CODE REVIEW

File: salt/modules/ddns.py [hunk 1]
```diff
@@ -191,16 +202,25 @@ def update(zone, name, ttl, rdtype, data, nameserver='127.0.0.1', timeout=5,
                     is_exist = True
                     break
 
+    log.debug('record exists: %s', is_exist)
```
Comment (dwoz) [inline salt/modules/ddns.py]: A bit of a nit-pick but can you make the `r` in `record` upper case please?

File: salt/modules/ddns.py [hunk 2]
```diff
@@ -191,16 +202,25 @@ def update(zone, name, ttl, rdtype, data, nameserver='127.0.0.1', timeout=5,
...
     dns_update = dns.update.Update(zone, keyring=keyring, keyname=keyname,
                                    keyalgorithm=keyalgorithm)
     if replace:
+        log.info('Replacing record %s', name)
         dns_update.replace(name, ttl, rdata)
     elif not is_exist:
+        log.info('Creating record %s', name)
         dns_update.add(name, ttl, rdata)
     else:
+        log.info('Not doing anything, record exists and in correct state', name)
```
Comment (dwoz) [inline salt/modules/ddns.py]: There is no string format for `name` here.

**Review by dwoz — CHANGES_REQUESTED** [review]

File: salt/modules/ddns.py (hunk 1 above)
Comment (DeForce) [reply salt/modules/ddns.py]: Yup! My bad :)

File: salt/modules/ddns.py (hunk 2 above)
Comment (DeForce) [reply salt/modules/ddns.py]: Now there is!

**Comment by DeForce** [comment]
@dwoz Changes are in, please check again! :)

**Comment by oeuftete** [comment]
I've re-submitted this against master in https://github.com/saltstack/salt/pull/58634.

**Pull request closed without merging**